processor/src/chiplets/hasher/tests.rs

```diff
@@ -404,6 +641,39 @@ fn check_hasher_state_trace(trace: &[Vec<Felt>], row_idx: usize, init_state: Has
     }
 }
 
+/// Makes sure that the trace is copied correctly on memoization
+fn check_memoized_trace(
+    trace: &[Vec<Felt>],
+    start_idx: usize,
+    end_idx: usize,
+    copied_start_idx: usize,
+    copied_end_idx: usize,
+) {
+    assert_eq!(end_idx - start_idx, copied_end_idx - copied_start_idx);
+
+    // make sure selector trace is copied correctly
+    let selector_trace = &trace[0..3];
+    for i in 0..(end_idx - start_idx) {
+        for j in 0..selector_trace.len() {
+            assert_eq!(
+                selector_trace[j][start_idx + i],
+                selector_trace[j][copied_start_idx + i]
+            )
+        }
+    }
+
+    // make sure hasher state trace is copied correctly
+    let hasher_state_trace = &trace[4..16];
+    for i in 0..(end_idx - start_idx) {
+        for j in 0..hasher_state_trace.len() {
+            assert_eq!(
+                hasher_state_trace[j][start_idx + i],
+                hasher_state_trace[j][copied_start_idx + i]
+            )
+        }
+    }
+}
```

Comment:
Similar comment here to above regarding refactoring and the use of magic numbers for indexing